HotelReservation/HotelReservation/ReservationSteps/StepsExecutor.cs

```diff
@@ -7,19 +7,19 @@ namespace HotelReservation.ReservationSteps
 {
     public class StepsExecutor
```

Comment:
To jest na obecną chwilę serce Twojego projektu, a nie posiada żadnego testu ;p Ta klasa wręcz musi być obtestowana. Jeśli z jakichś powodów jest ciężko coś przetestować to jest hint, że należy klasę lekko przeorganizować.

Reply:
Fixed - nie jestem pewien co do testowania wywołania stepu, który nie jest obsługiwany przez kod. Na tę chwilę dodałem TestStep do Enuma ReservationStepType.

Reply:
No i tutaj zaczyna się testowanie w izolacji.
ShouldNotThrowExceptionForValidReservationStepsList
ShouldThrowExceptionForNotValidReservationStepsList

To są testy, które dotyczą factory. Niech ona sobie rzuci wyjątek jak nie potrafi stworzyć stepu. Co zresztą robi, ale testu na to nie ma - dodaj proszę.
ReservationStepType.TestStep - coś takiego nie ma sensu w kodzie produkcyjnym, dlatego nie możemy dodać tego na potrzeby testów. Enum to jest pod spodem numerek. Można tego użyć do testowania przypadków, gdy jakaś wartość jest nieobsługiwana. Rzutowanie -1 na enum powinno wystaczyć:

_subject.CreateInstance((ReservationStepType)(-1)).Should().ThrowException()

Handlowanie wyjątków w StepExecutor nie jest jakoś super potrzebne wg mnie. Message w wyjątku jest bardziej dla programisty, niż dla usera. Tak naprawdę połykasz informacje przez co trudniej będzie się dowiedzieć gdzie wystąpił błąd.

Wracając do testów:
* Przypadki testowe mają odzwierciedlać odpowiedzialność klasy. Jakbym Cię zapytał co ta klasa robi, to powiesz, że rzuca wyjątek lub nie? Nie brakuje Ci tutaj nic? 
* Chcielibyśmy testować klasy w oderwaniu od innych klas. Obecnie testujesz tą klasę razem z kodem factory. Słowa kluczowe do poszukania: mock/stub

Jeśli będziesz miał problemy z przetestowaniem tej klasy w odizolowaniu od factory - zagadaj mnie to Ci trochę więcej podpowiem

Reply:
> No i tutaj zaczyna się testowanie w izolacji.
> ShouldNotThrowExceptionForValidReservationStepsList
> ShouldThrowExceptionForNotValidReservationStepsList
> 

> Wracając do testów:
> 
> * Przypadki testowe mają odzwierciedlać odpowiedzialność klasy. Jakbym Cię zapytał co ta klasa robi, to powiesz, że rzuca wyjątek lub nie? Nie brakuje Ci tutaj nic?
> * Chcielibyśmy testować klasy w oderwaniu od innych klas. Obecnie testujesz tą klasę razem z kodem factory. Słowa kluczowe do poszukania: mock/stub
> 
> Jeśli będziesz miał problemy z przetestowaniem tej klasy w odizolowaniu od factory - zagadaj mnie to Ci trochę więcej podpowiem

Reszta, poza wymienionym wyżej - fixed.

Zastanawiam się, czy użycie biblioteki "Moq" do stworzenia mocków/stubów jest najlepszym rozwiązaniem (dodałem ją już do projektu). Możemy chwilę o tym pogadać.